### DISCUSSION TITLE
No windows available for training"

### DISCUSSION THREAD

**Discussion by hiandersson** [Nixtla/neuralforecast#573] [Q&A]
When I train models, especially TFT, I sometimes get this exception "No windows available for training" while optimising hyper parameters. I look in the code and its not super easy to tell what is going on ..
```
           # Protection of empty windows
            if final_condition.sum() == 0:
                raise Exception("No windows available for training")
```
Has it something to do with time series of a unique id being smaller than encoder length or is it related to something else?

**Comment by kdgutier** [comment]
Hi @hiandersson,

All the WindowsBased models (MLP, NBEATS, NHITS, TFT,...) use the `create_windows` method to train and predict.
https://github.com/Nixtla/neuralforecast/blob/main/neuralforecast/common/_base_windows.py#L178

You might want to limit the AutoTFT configuration space by limiting the `input_size` to match the length of your series.

**Reply by mak-rayate** [reply]
I'm encountering the same exception error repeatedly. @kdgutier, would you be able to provide a sample example? I've searched extensively through the documentation, but I couldn't find anything related to it.

**Comment by aasyedd1** [comment]
Hi. I am encountering the same problem. My time series is quite simple with 1440 rows and 5 columns. It consists of hourly readings of load consumption data. I am trying to implement simple example from the tutorial 
https://nixtlaverse.nixtla.io/neuralforecast/examples/exogenous_variables.html and the model is given as 

```
horizon = 24 # day-ahead daily forecast
models = [NHITS(h = horizon,
                input_size = 5*horizon,
                start_padding_enabled = True,
                futr_exog_list = ['temperature', 'hour', 'weekday'], # <- Future exogenous variables
                scaler_type = 'robust'),
                BiTCN(h = horizon,
                input_size = 5*horizon,
                start_padding_enabled = True,
                futr_exog_list = ['temperature', 'hour', 'weekday'], # <- Future exogenous variables
                scaler_type = 'robust',
                )]
```
And forecasting is done by 
`
nf = NeuralForecast(models=models, freq='H')
nf.fit(df=df)
`
Any help in this regard would be appreciated

**Comment by aasyedd1** [comment]
The issue has been solved. it is related to incorrect `unique_id`. Be sure that time series has enough samples in `unique_id`. I incorrectly referred it index number and thus the code assumes that there are as many times series as the number of rows in dataset.

**Reply by mgeracie** [reply]
As of what version has this issue been fixed?

**Comment by rjguedes8** [comment]
I'm using TiDE and for some combination of hyper parameters I face the same issue.
My series have a length >= input size + h, any idea why does this happen? I cannot find anything common between the hyper parameters that failed that might be triggering this exception...

these do not work:
```
{"input_size": 12, "h": 13, "num_encoder_layers": 18, "num_decoder_layers": 11, "decoder_output_dim": 16, "hidden_size": 19, "temporal_width": 20, "temporal_decoder_dim": 64, "dropout": 0.2201250055472177, "batch_size": 16, "max_steps": 2887, "random_seed": 42, "learning_rate": 0.0036755406235537713, "layernorm": false, "scaler_type": "standard", "num_lr_decays": 7, "val_check_steps": 100, "start_padding_enabled": true}
{"input_size": 12, "h": 13, "num_encoder_layers": 3, "num_decoder_layers": 6, "decoder_output_dim": 6, "hidden_size": 8, "temporal_width": 31, "temporal_decoder_dim": 32, "dropout": 0.167877884832288, "batch_size": 32, "max_steps": 1493, "random_seed": 42, "learning_rate": 0.0012947352550811552, "layernorm": false, "scaler_type": "minmax", "num_lr_decays": 7, "val_check_steps": 50, "start_padding_enabled": true}
{"input_size": 26, "h": 13, "num_encoder_layers": 5, "num_decoder_layers": 7, "decoder_output_dim": 26, "hidden_size": 14, "temporal_width": 1, "temporal_decoder_dim": 64, "dropout": 0.2447828048783552, "batch_size": 32, "max_steps": 1183, "random_seed": 42, "learning_rate": 0.012868199705960067, "layernorm": false, "scaler_type": "standard", "num_lr_decays": 7, "val_check_steps": 50, "start_padding_enabled": true}
{"input_size": 12, "h": 13, "num_encoder_layers": 30, "num_decoder_layers": 27, "decoder_output_dim": 26, "hidden_size": 28, "temporal_width": 6, "temporal_decoder_dim": 16, "dropout": 0.22650625477678882, "batch_size": 32, "max_steps": 2261, "random_seed": 42, "learning_rate": 0.04951797346057748, "layernorm": false, "scaler_type": "robust", "num_lr_decays": 3, "val_check_steps": 100, "start_padding_enabled": true}
```

this works:
```
{"input_size": 52, "h": 13, "num_encoder_layers": 9, "num_decoder_layers": 6, "decoder_output_dim": 10, "hidden_size": 24, "temporal_width": 13, "temporal_decoder_dim": 32, "dropout": 0.21459933827060712, "batch_size": 32, "max_steps": 1121, "random_seed": 42, "learning_rate": 0.007506012394677122, "layernorm": false, "scaler_type": "standard", "num_lr_decays": 3, "val_check_steps": 50, "start_padding_enabled": true}
```
